sdk/src/spectrumx/client.py

```diff
@@ -484,5 +485,87 @@ def upload_capture(
         else:
             return capture
 
+    def _handle_existing_capture_error(
+        self, err: SDSError, captures: list[Capture]
+    ) -> bool:
+        """
+        Handle the case where a capture already exists.
+        Returns True if handled, False otherwise.
+        """
+        if "drf_unique_channel_and_tld" not in str(err).lower():
+            return False
+        try:
+            error_msg = str(err)
+            uuid_start = error_msg.find("another capture: ") + len("another capture: ")
+            uuid_end = error_msg.find(" ", uuid_start)
+            if uuid_end == -1:
+                uuid_end = len(error_msg)
+            existing_uuid_str = error_msg[uuid_start:uuid_end].strip()
+            existing_uuid = UUID(existing_uuid_str)
+            existing_capture = self.captures.read(capture_uuid=existing_uuid)
+            captures.append(existing_capture)
+        except (SDSError, ValueError, IndexError):
+            return False
+        else:
+            return True
+
+    def upload_multichannel_drf_capture(
+        self,
+        *,
+        local_path: Path | str,
+        sds_path: PurePosixPath | Path | str = "/",
+        index_name: str = "",
+        channels: list[str],
+        verbose: bool = True,
+        warn_skipped: bool = False,
+        raise_on_error: bool = True,
+    ) -> list[Capture] | None:
+        """Uploads a capture with multiple channels by running `upload`
+        once for the whole directory. Then, it creates a capture for each channel.
+
+        *Note: This method is only for DigitalRF captures.*
+        """
+        upload_results = self.upload(
+            local_path=local_path,
+            sds_path=sds_path,
+            verbose=verbose,
+            warn_skipped=warn_skipped,
+        )
+
+        if not process_upload_results(
+            upload_results,
+            raise_on_error=raise_on_error,
+            verbose=verbose,
+        ):
+            return None
+
+        captures: list[Capture] = []
+        for channel in channels:
```

Comment:
I think we can issue a user warning if the `channels` list is empty (thus returning an empty capture list after the file upload)